google/cloud/storage/_opentelemetry_meter.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Manages OpenTelemetry metrics creation and handling"""
+
+try:
+    from opentelemetry import metrics
+    from opentelemetry.sdk.metrics import Counter
+    from opentelemetry.exporter.cloud_monitoring import CloudMonitoringMetricsExporter
+
+    meter_provider = metrics.get_meter_provider()
+
+    # If opentelemetry packages have been installed and a MeterProvider setup
+    #   we are ready to use opentelemetry
+    OPENTELEMETRY_READY = not isinstance(meter_provider, metrics.DefaultMeterProvider)
+except ImportError:
+    OPENTELEMETRY_READY = False
+
+FUNCTION_NAME_KEY = "instrumented_function_name"
+
+if OPENTELEMETRY_READY:
+
+    meter = metrics.get_meter(__name__)
+    meter_provider.start_pipeline(meter, CloudMonitoringMetricsExporter(), interval=15)
```

Comment:
This is likely out of scope for this PR, but would be valuable to address, perhaps in the design doc:
What if the GCS customer in question does not want to export directly for Cloud Monitoring?
-> they may be using a 3rd party monitoring tool, or they may want to export to the OpenTelemetry collector (for performance or security reasons).

Does (or should) OpenTelemetry offer any ways of doing this?

Reply:
So because in this case, we do always want to export to Cloud Monitoring we'll keep this code. The user can just add in their own code
`meter_provider.start_pipeline(meter, OtherExporter(), interval=15)`
To make it export to another backend in addition to Cloud Monitoring.

Reply:
Where would users add that code? This is a private initializer inside this library. I wouldn't expect users to want to fork and modify this library. I believe exporting to the collector will be very common, especially in security-hardened deployments where you don't want components to be able to make arbitrary network connection to the internet.

Reply:
Sorry, to be clear it would look like
```
# Imports
meter_provider = MeterProvider(stateful=False)
metrics.set_meter_provider(meter_provider)
meter_provider.start_pipeline(otherexporter()...)

from google.cloud import storage
# user's logic
```
Note that the only new line is `meter_provider.start_pipeline(otherexporter()...)`

Reply:
Summary from off-line conversation: The current implementation is good: This is a Cloud Monitoring-only integration and the users opt in by installing the Cloud Monitoring exporter themselves. The larger question is whether we should allow the customer to configure their own exporter. That is not supported by the OpenTelemetry API and thus out of scope for this PR.